scripts/lib/CIME/BuildTools/cmakemacroswriter.py

```diff
@@ -117,3 +118,42 @@ def end_ifeq(self):
         """
         self.indent_left()
         self.write_line("endif()")
+
+    def _transform_value(self, name, value):
+        """Some elements need their values transformed in some way for CMake to handle them properly.
+        This method does those transformations.
+
+        Args:
+        - name (str): name of element
+        - value (str): value of element
+
+        Returns transformed value
+        """
+
+        value_transformed = value
+        if self._element_needs_whitespace_removal(name):
+            value_transformed = value_transformed.strip()
+
+        return value_transformed
+
+    def _element_needs_whitespace_removal(self, name):
```

Comment:
Do we really need this list?  Wouldn't it be enough to just always remove trailing (and leading) white space?